src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp

```diff
@@ -3025,23 +3025,12 @@ HRESULT STDMETHODCALLTYPE CorProfiler::JITCachedFunctionSearchStarted(FunctionID
 
     if (!has_loader_injected_in_appdomain)
     {
+        Debug("Disabling NGEN due missing loader...");
         // The loader is missing in this AppDomain, we skip the NGEN image to allow the JITCompilationStart inject it.
         *pbUseCachedFunction = false;
         return S_OK;
     }
 
-    RejitHandlerModule* handlerModule = nullptr;
```

Comment:
How did this fix issue? It looks like we only set `*pbUseCachedFunction = false;` when the loader is missing so I don't understand how this works

Reply:
Yes, I found that RequestReJIT is doing that work already, so we only need to ReJIT an NGEN method to do the rewriting. If I set the * pbUseCachedFunction flag to false that triggers a normal JITCompilation with the original IL that replaces our rewrite. That’s why we weren’t instrumenting in that case.

Reply:
Oh I see, so we want to set the flag to true and let RequestReJIT take care of the rewriting. That makes sense, thanks for explaining!

Reply:
That’s right and keep it disabled until the loader inject itself in the JITCompilationStarted callback.